src/store.c

```diff
@@ -0,0 +1,71 @@
+#include "common.h"
+#include <sys/stat.h>
+
+/*
+ * This function creates a folder.
+ * The folder name is the name of the database
+ */
+void createDatabase(struct ParseTree* createCommand) {
+
+  assert(createCommand->commandType == CREATE_DATABASE,
+      "Incompatible parseTree commandType to function createDatabase");
+  mkdir(DATABASE_DIR, 0700);
+
+  char folderPath[NAME_LIMIT];
+  getcwd(folderPath, sizeof(folderPath));
+
+  sprintf(folderPath, "%s/%s", DATABASE_DIR, createCommand->table);
+  mkdir(folderPath, 0700);
+
+
+  setDatabase(createCommand->table);
+}
+
+/*
+ * Sets the global currentDatabase variable
+ */
+void setDatabase(char* database) {
+  if(currentDatabase == NULL) {
+    currentDatabase = calloc(1, NAME_LIMIT);
+  }
+  snprintf(currentDatabase, NAME_LIMIT,
+     "%s", database);
+}
+
+/**
+ * Creates the table structure on disk.
+ * Assumes that field list is NULL terminated.
+ */
+void createTable(struct ParseTree* parseTree) {
+
+  // validations!
+  assert(parseTree->commandType == CREATE_TABLE,
+    "Expectd parseTree commandType to be CREATE_TABLE");
+  assert(parseTree->fields, "Fields must be provided!");
+  assert(parseTree->table, "Table name must be provided");
+  assert(currentDatabase, "Please select a database!");
+
+  // Create a table file in the default database directory
+  char tablePath[1000];
+  snprintf(tablePath, sizeof(tablePath), "%s/%s/%s", DATABASE_DIR, currentDatabase, parseTree->table);
+  FILE* tableFile = fopen(tablePath, "w");
+
+  int i = 0;
+
+  struct Field* fields[FIELD_LIMIT];
+  char headerLine[FIELD_LIMIT*NAME_LIMIT+FIELD_LIMIT];
+  char tmpLine[FIELD_LIMIT*NAME_LIMIT+FIELD_LIMIT];
+  struct Field* providedFields = parseTree->fields;
+
+  while((*(parseTree->fields+i)).name != NULL) {
+    char name[NAME_LIMIT] = "";
+    char type = parseTree->fields[i].fieldType;
+    snprintf(name, sizeof(name), parseTree->fields[i].name);
+    snprintf(tmpLine, sizeof(headerLine), headerLine);
+    snprintf(headerLine, sizeof(headerLine), "%s|%s[%c]", tmpLine, name, type);
```

Comment:
So it turns out this is an undesirable way to write records.

Will have to write an adapter to force all lines to be the same size! For an update command, we can overwrite the record in place, but also avoid this problem: http://stackoverflow.com/questions/508983/how-to-overwrite-some-bytes-in-the-middle-of-a-file-with-python